src/UI/Component/Table/Presentation.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Table;
+
+/**
+ * This describes a Presentation Table
+ */
+interface Presentation extends \ILIAS\UI\Component\Component {
+
+	/**
+	 * Get a table like this with title $title.
+	 *
+	 * @param string 	$title
+	 * @return \Presentation
+	 */
+	public function withTitle($title);
+
+	/**
+	 * Get the title of the table.
+	 *
+	 * @return string
+	 */
+	public function getTitle();
+
+	/**
+	 * Get a table like this with these view controls.
+	 *
+	 * @param \ViewControl[] 	$view_controls
+	 * @return \Presentation
+	 */
+	public function withViewControls(array $view_controls);
+
+	/**
+	 * Get view controls to be shown in the header of the table.
+	 *
+	 * @return ILIAS\UI\Component\ViewControl[]
+	 */
+	public function getViewControls();
+
+	/**
+	 * Get a table like this with the closure $row_mapping.
+	 * The closure MUST accept the following parameter
+	 *   \PresentationRow $row
+	 *   mixed $record
+	 *   \Factory $ui_factory
+	 *   mixed $environment
+	 *
+	 * The closure MUST return \PresentationRow
+	 *
+	 * @param \Closure 	$row_mapping
+	 * @return \Presentation
+	 */
+	public function withRowMapping(\Closure $row_mapping);
```

Comment:
What is this used for?

Reply:
@klees I see the point here, but a Closure is quite hard to describe and even harder to test. 
under the premise we get PHP7 for ILIAS 5.4, how about using anonymous classes here? then you could define the interface of a row-mapping.

Reply:
...and it would be possible to deliver some basic row-mappers.

Reply:
Yeah, anonymus class would be perfectly fine.